mlir/include/mlir/Interfaces/LoopLikeInterface.td

```diff
@@ -235,6 +231,35 @@ def LoopLikeOpInterface : OpInterface<"LoopLikeOpInterface"> {
   }];
 
   let extraSharedClassDeclaration = [{
+    /// If there is a single induction variable return it, otherwise return
+    /// std::nullopt.
+    ::std::optional<::mlir::Value> getSingleInductionVar() {
+      if (this->getInductionVars().size() == 1)
```

Comment:
this code will create a temporary `SmallVector` just to get size and then another one on the next line, better to cache.